activation/activation.go

```diff
@@ -765,13 +761,8 @@ func (b *Builder) discardChallengeIfStale() bool {
 	return false
 }
 
-// SignAtx signs the atx with specified signer and assigns the signature into atx.Sig
-// this function returns an error if atx could not be converted to bytes.
-func SignAtx(signer signer, atx *types.ActivationTx) error {
-	bts, err := atx.InnerBytes()
-	if err != nil {
-		return fmt.Errorf("inner bytes of ATX: %w", err)
-	}
-	atx.Sig = signer.Sign(bts)
-	return nil
+// SignAndFinalizeAtx signs the atx with specified signer and calculates the ID of the ATX.
+func SignAndFinalizeAtx(signer signer, atx *types.ActivationTx) error {
+	atx.Sig = signer.Sign(atx.SignedBytes())
+	return atx.CalcAndSetID()
```

Comment:
no. 

for ATX builder workflow, nodeID is baked in in types.NewActivationTx(). there should not be calls to SetNodeID.
`SetNodeID` should only be called in the following codepath:
- receive an ATX from gossip
- read an ATX from database

Reply:
Perhaps there could be a dedicated function like:
```go
func AtxFromBytes(data []byte) (*types.ActivationTx, error) {
  // deserialize
  // verify signature
  // calc and set ID
  return atx, nil
}
```

and remove `CalcAndSetID()` function at all?

Reply:
what you are describing is can be done for the gossip flow. 

but for ATX builder, we still need to sign and calculate the ID. and since ID is a private field in the common/types pkg, there still need to be a public method for setting ID. we could calculate it outside and call SetID, but it feel superficial. 

for the database flow we don't verify signature again after getting the raw byte. that's why we need SetNodeID there.
to be consistent we can also call types.ActiviationTx() from there with necessary data. 

in general i am not very happy with current set of methods (SetID, CalcAndSetID, CalcAndSetNodeID). i feel like they can somehow be simplified. but i want to do that in a separate PR later.